core/vdbe/datetime.rs

```diff
@@ -828,57 +832,35 @@ mod tests {
         ];
 
         for (input, expected) in test_cases {
+            thread::sleep(Duration::from_millis(50));
+
             let result = exec_time(&[input]);
             if let OwnedValue::Text(result_str) = result {
-                assert_eq!(result_str.as_str(), expected);
+                println!("Comparing: result={} expected={}", result_str, expected);
+
+                let result_time = NaiveTime::parse_from_str(&result_str, "%H:%M:%S")
+                    .expect("Failed to parse result time");
+                let expected_time = NaiveTime::parse_from_str(expected, "%H:%M:%S")
+                    .expect("Failed to parse expected time");
+
+                // Allow a difference of up to 3 seconds
```

Comment:
This should always be deterministic and we are allowing second differences which is hard to reason about. All of the test cases have a fixed Value that must be always be parsed to the same expected value. The *only* case where this might not happen is with `now` which in this case it could maybe make sense. 

Basically I suggest we test `now` in another test isolated where the undeterminism happens and leave the rest of the test untouched. What do you think?